lms/djangoapps/course_home_api/progress/v1/serializers.py

```diff
@@ -1,9 +1,62 @@
 """
 Progress Tab Serializers
 """
-
+import uuid
 from rest_framework import serializers
 from lms.djangoapps.course_home_api.outline.v1.serializers import CourseBlockSerializer
+from rest_framework.reverse import reverse
+
+
+class CoursewareSummarySerializer(serializers.Serializer):
+    """
+    Serializer
+    """
+    blocks = serializers.SerializerMethodField()
+
+    def get_blocks(self, blocks):
```

Comment:
So looking at a lot of this, it looks like you are just doing a straight mapping in a lot of these cases where maybe it will be better for us to use a sub-serializer that will be able to also add in some validation since it is using Serializer Fields.